pkg/kubemark/hollow_kubelet.go

```diff
@@ -91,7 +92,7 @@ func (hk *HollowKubelet) Run() {
 	if err := kubeletapp.RunKubelet(&options.KubeletServer{
 		KubeletFlags:         *hk.KubeletFlags,
 		KubeletConfiguration: *hk.KubeletConfiguration,
-	}, hk.KubeletDeps, false); err != nil {
+	}, hk.KubeletDeps, false, types.NodeName("")); err != nil {
```

Comment:
To me, it would be more readable for us to pass `nil` instead of the empty string when we don't know the node name. Is that an option?

Reply:
That's great , i got it

Reply:
> To me, it would be more readable for us to pass `nil` instead of the empty string when we don't know the node name. Is that an option?

Hello, This problem is very confusing to me. If I pass in the nil type directly, then the previous function input parameter needs a lot of adjustment. Before the nodeName type is a string. Is there a more elegant way to deal with this problem?

Reply:
> To me, it would be more readable for us to pass `nil` instead of the empty string when we don't know the node name. Is that an option?


Inside the call path of the RunKubelet method, there are many places that repeatedly calculate the properties for nodeName，such as 

-  [L990](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L990)

- [L385](https://github.com/kubernetes/kubernetes/blob/11678fb1c029ee1f3c7223e87d339b2d0016fdb8/pkg/kubelet/kubelet.go#L385)

And all of them have the same calculate path HostnameOverride->hostName->nodeName. 

Is there a better way to handle this problem ， such as  design a structure that performs an assignment at the initial call. Have edit rights through a unified method. Reduce redundant code and improve cohesion
just like this 
```
type NodeName struct {
	HostnameOverride string
	HostName string
	NodeName string
}
```

Reply:
Good observation!

Are all of those calls occurring within the kubelet? I wonder if we could store the data on the kubelet, and then add an accessor method that returns the data if its already been calculated, and calculates and returns it if not?

Reply:
> Good observation!
> 
> Are all of those calls occurring within the kubelet? I wonder if we could store the data on the kubelet, and then add an accessor method that returns the data if its already been calculated, and calculates and returns it if not?

In addition to kubelet there are many other places to call this field，but only this one entry is assigned. Do you have any better suggestions?